catch_analysis_tools/astrometry.py

```diff
@@ -0,0 +1,435 @@
+    telescope_image_sub : np.ndarray
+        Background-subtracted image array.
+    """
+    mask_lower_limit = 0 
+    mask = np.isinf(image) | (image <= mask_lower_limit)
+    image = np.ma.masked_array(image, mask)
+    image_data = np.asarray(
+        getattr(image, "filled", lambda x: x)(0),
+        dtype=np.float32
+    )
+    bkg = sep.Background(image_data)
+    telescope_image_sub = image_data - bkg.back()
+    sep.set_sub_object_limit(500)
+    sources = sep.extract(
+        telescope_image_sub,
+        thresh=snr,
+        err=bkg.globalrms,
+        deblend_nthresh=16
+    )
+    source_list = pd.DataFrame(sources)
+    flux, flux_err, _ = sep.sum_circle(
+        telescope_image_sub,
+        source_list['x'], source_list['y'],
+        aperture_radius,
+        err=bkg.globalrms
+    )
+    source_list['aperture_sum'] = flux
+    source_list['aperture_err'] = flux_err
+    source_list = source_list[source_list['aperture_sum'] > 0].reset_index(drop=True)
+    return source_list, telescope_image_sub
+
+
+def load_wcs(output_wcs):
+    """
+    Load a WCS solution from a FITS file header.
+
+    Parameters
+    ----------
+    output_wcs : str
+        Path to the FITS file containing the WCS header from astrometry.net().
+
+    Returns
+    -------
+    wcs_solution : astropy.wcs.WCS
+        World coordinate system solution object.
+    """
+    if not os.path.exists(output_wcs):
+        raise FileNotFoundError(f"WCS file not found: {output_wcs}")
+    with fits.open(output_wcs) as hdul:
+        wcs_solution = WCS(hdul[0].header)
+    return wcs_solution
+
+
+def retrieve_sources(source_list, wcs_solution):
+    """
+    Convert pixel coordinates to sky coordinates using a WCS.
+
+    Parameters
+    ----------
+    source_list : pd.DataFrame
+        Table with 'x' and 'y' pixel positions of detected sources.
+    wcs_solution : astropy.wcs.WCS
+        World coordinate system solution object.
+
+    Returns
+    -------
+    source_list : pd.DataFrame
+        Updated table including 'RA' and 'Dec' columns in degrees.
+    sky_coords : astropy.coordinates.SkyCoord
+        SkyCoord object with celestial coordinates of sources.
+    """
+    world = wcs_solution.pixel_to_world(source_list['x'], source_list['y'])
+    source_list['RA'] = [c.ra.deg for c in world]
+    source_list['Dec'] = [c.dec.deg for c in world]
+    sky_coords = SkyCoord(source_list['RA'], source_list['Dec'], unit='deg')
+    return source_list, sky_coords
+
+
+def calibrate_photometry(sky_coords, source_list, catalog='PanSTARRS1'):
+    """
+    Calibrate instrumental magnitudes against a Pan-STARRS1 catalog.
+
+    Parameters
+    ----------
+    sky_coords : astropy.coordinates.SkyCoord
+        Celestial coordinates of detected sources.
+    source_list : pd.DataFrame
+        Table of detected sources containing 'aperture_sum'.
+    catalog : str, optional  
+        Name of the photometric catalog to use (default is 'PanSTARRS1').  
```

Comment:
We also need a parameter that is the photometric bandpass to calibrate to, e.g., "g", "r", "i", etc..